trio/_core/_tests/tutil.py

```diff
@@ -98,9 +100,7 @@ def restore_unraisablehook() -> Generator[None, None, None]:
 
 # template is like:
 #   [1, {2.1, 2.2}, 3] -> matches [1, 2.1, 2.2, 3] or [1, 2.2, 2.1, 3]
-def check_sequence_matches(
-    seq: Sequence[object], template: Iterable[object | set[object]]
-) -> None:
+def check_sequence_matches(seq: Sequence[T], template: Iterable[T | set[T]]) -> None:
```

Comment:
Hm. I'm not sure I quite follow why the `T` here -- but I don't remember the variance of `Iterable` so maybe this makes sense... Though I guess maybe for `set`.... aaa whatever, might as well ask.

Reply:
I.. don't quite follow why the signature would be confusing? Though the comment above reversing them might be the cause of confusion.
You pass it a sequence of some thing, e.g. `[1, 2, 3, 4, 5]` or `[1, 2, 3, 5, 4]`, and use this function to see whether that matches a template - which can include either objects of the same type, or if you expect the order to sometimes be jumbled you can specify them inside a set*: `[1, 2, 3, {4, 5}]`. So the template should be an `Iterable` which contains objects of the same type your sequence consists of, or sets of that object. Variance is not related in any way.

The `Sequence` vs `Iterable` is because the function does slicing on `seq`, but only iterates over `template`.

\* I did not expect this to be how the function actually worked, and wrote an incorrect description at first where I assumed the sets were about unions. Not a great name imho.

Improved the comments accompanying the function.

Reply:
Well the `T` w/ `Sequence` (covariant) means that something like this would pass typechecking: `check_sequence_matches([5.0, 4.0], [5, 4])` but I was in fact being blind to the fact that `object` has that problem too.

Reply:
Ah, huh. Yeah I don't think we really care about that level of precision, as it's only really used in a few tests and not a public function. I need to get better at variance, for sure.